vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleOverviewComponent.java

```diff
@@ -352,6 +371,48 @@ private void setupSearchFieldForExperimentTabs(String experimentName,
         }
       });
     }
+
+
+    private void processBatchRegistration(BatchRegistrationContent batchRegistrationContent) {
+      //Todo add Batch name here and trigger processSampleCreation() method
+      Result<BatchId, BatchRegistrationService.ResponseCode> batch = batchRegistrationService.registerBatch(
+          batchRegistrationContent.batchLabel(), batchRegistrationContent.isPilot());
+      batch.onValue(result -> {
+        batchRegistrationDialog.resetAndClose();
+        lastCreatedBatch = result;
+        //ToDo Replace Values
+      }).onError(e -> {
+        //ToDo What should happen here?
+      });
+    }
```

Comment:
Todos need to be either addressed, or removed :)

Suggestion for method refactoring: `processing` is a technical term, we want to avoid that. What is happening is, that this methods calls the service and commands the batch registration. So `registerBatch` is fine. I would exclude the result handling to the point where this method is called, otherwise your method is doing two things, and we want methods only do one thing.